custom/enikshay/integrations/bets/repeaters.py

```diff
@@ -411,9 +427,25 @@ class BETSLocationRepeater(BETSRepeaterMixin, LocationRepeater):
     )
 
     def allowed_to_forward(self, location):
+        # if this location is already in the repeater queue, don't forward again
+        if queued_payload(location.domain, location.location_id):
+            return False
+
+        # If this location has already been forwarded without any changes, then
+        # don't send it to BETS again
+        successful_records = location.metadata.get('BETS_location_repeat_record_ids')
+        if successful_records:
+            latest_record_id = successful_records.split(" ")[-1]
```

Comment:
Whats the reason for keeping all attempts and not just the last successful record?

Reply:
I guess we could just save the last one. I figure this will make it easier to pull all the history from the DB if ever we are trying to debug why certain records are sending incorrectly or something, but maybe thats a YAGNI.

Reply:
Ya. One can also use `get_repeat_records_by_payload_id` and then their attempts to get all attempts when needed to pull history.

Reply:
yeah, this way allows you to do it without having to start a shell session (using raw_doc).